raster/src/main/scala/geotrellis/raster/ArrayMultibandTile.scala

```diff
@@ -428,4 +428,25 @@ class ArrayMultibandTile(_bands: Array[Tile]) extends MultibandTile with MacroMu
 
     new ArrayMultibandTile(newBands)
   }
+
+  override def equals(other: Any) = {
+    if (other == null) {
+      false
+    } else other match {
+      case that : ArrayMultibandTile =>
+        case class NotEqualException() extends Exception()
+        var result = true
+        try {
+          for (b <- 0 until bandCount) {
+            if (band(b) != that.band(b)) throw NotEqualException()
+          }
+        } catch {
+          case e : NotEqualException =>
```

Comment:
We don't use exceptions for control flow. That's considered a "Bad Thing" in Scala.

You could fold:

```scala
val result = 
  (0 until bandCount)
    .foldLeft(true) { (acc, b) => acc && (band(b) == that.band(b)) }
```

That is doing a foreach over integer values, and while the size should be small, the actually fastest way to do it is not as pretty (because of the var):

```scala
var result = true
cfor(0)(_ < bandCount, _ + 1) { b =>
  result = result && (band(b) == that.band(b))
}
```